metricflow/dataflow/builder/node_evaluator.py

```diff
@@ -283,10 +298,24 @@ def _find_joinable_candidate_nodes_that_can_satisfy_linkable_specs(
                             join_on_partition_dimensions=join_on_partition_dimensions,
                             join_on_partition_time_dimensions=join_on_partition_time_dimensions,
                             validity_window=validity_window_join_description,
-                            join_type=join_type,
+                            join_type=default_join_type,
                         )
                     )
 
+            # If right node is time spine source node, we can cross join.
+            if right_node == time_spine_source_node:
+                needed_metric_time_specs = LinkableSpecSet.from_specs(needed_linkable_specs).metric_time_specs
+                candidates_for_join.append(
+                    JoinLinkableInstancesRecipe(
+                        node_to_join=right_node,
+                        join_on_entity=None,
+                        satisfiable_linkable_specs=list(needed_metric_time_specs),
+                        join_on_partition_dimensions=(),
+                        join_on_partition_time_dimensions=(),
+                        join_type=SqlJoinType.CROSS_JOIN,
+                    )
+                )
```

Comment:
I feel like this belongs at the top of the loop, and with a `continue` here to indicate that this is a short-circuit.

Should we ever have the time spine where it's valid to go through the rest of the logic in this loop over the set of possible `right nodes`? If not let's move this up and make the special case around time spine handling more obvious.

Reply:
Will do!